### ISSUE TITLE
is it possible to set the row color by a row.field value?

### ISSUE THREAD

**Issue by aufgussmeister** [jbetancur/react-data-table-component#293]
I would like to have different color of row (green for completed work, white for incompleted work). is there a way to do that?

Thanks

**Comment by jbetancur** [comment]
Currently no. But such a feature would also need to account for font color as well as hover color and even different cases (completed, incomplete, error, pending etc..). We would need to pass in a style object of sorts.

An idea would be to allow you to define some row formatting on your row data. For example, `rowStyleField` could be a new Object property that you can pass some css-in-js compliant object to. 

Internally, RDT uses styled-components which allows for style objects (i.e. a css-in-js compliant object0
https://www.styled-components.com/docs/advanced#style-objects

Of course, it would be on you to map or optimize your data from the source so that the colors reflect some boolean property you are using to designate as whatever status.

e.g. you can map your data like this:

```js
const myFormattedRows = myData.map(datum => {
 const myRowStyle= {};

 if (datum.status === 'completed') {
    myRowStyle.backgroundColor ='green';
    myRowStyle.color = 'white';
 }

return { ...datum, myRowStyle };
});
```

Where your data looks like this now:
```js
[{ 
 id: 1, 
 name: 'my row',
 status: 'completed',
 myRowStyle: {
  backgroundColor: 'green',
  color : 'white',
 ...
 }',
...
}]
```

```js
<DataTable 
 ....
 data={myFormattedRows }
 rowStyleField="myRowStyle"
/>
```


I'll give this some more thought as it could be an interesting feature.

**Comment by jbetancur** [comment]
This feature is now available in 3.8.0! see README and storybook examples

Similar to columns you can now pass in an array of expressions as `conditionalRowStyles` that will then apply a style if that expression evaluates to true.

The style would just be in css-in-js style objects format. 

```js
const conditionalRowStyles= [
  {
    when: row => row.completed,
    style: {
      backgroundColor: 'rgba(63, 195, 128, 0.9)',
      color: 'white',
      '&:hover': {
        cursor: 'pointer',
      },
    },
  },
];

...
<DataTable ... conditionalRowStyles={conditionalRowStyles} />
```

Originally, I thought of just mapping some style onto the data, but I don't like the idea of having to pollute the input data with styles, furthermore, mapping your data in that way probably has a performance impact for larger data sets.

With `conditionalRowStyles`, any performance impact scales linearly(n) on how many expressions are in your `rowStyleExpressions` . e.g. I doubt having 10 expressions be noticeable but I don't foresee people using more than 2 or 3 (that would make for some ugly ui)
